variety/VarietyWindow.py

```diff
@@ -1376,6 +1377,77 @@ def apply_filters(self, to_set, refresh_level):
             logger.exception(lambda: "Could not apply filters:")
             return to_set
 
+    def apply_auto_rotate(self, to_set):
+        try:
+            if self.options.wallpaper_auto_rotate:
+                target_file = os.path.join(
+                    self.wallpaper_folder, "wallpaper-auto-rotated-%s.jpg" % Util.random_hash()
+                )
+                cmd = "convert %s -auto-orient %s" % (shlex.quote(to_set), shlex.quote(target_file))
+                logger.info(lambda: "ImageMagick auto-rotate cmd: " + cmd)
+                cmd = cmd.encode("utf-8")
+
+                result = os.system(cmd)
```

Comment:
I'd recommend doing `subprocess.call()` on an array of command arguments; this way you don't have to worry about manually quoting things.

e.g. `cmd = ["convert", to_set, "-auto-orient", target_file]`

Reply:
Agree, but this is currently consistent with the other similar places whee we execute convert commands, and thus more suitable to eventually rework all the code duplication that is around these ugly parts. Moving to `subprocess.call` in the other places would be non-trivial - our filter configs are string-based currently, so it's actually easier to build a string command than to split into an array.

As a whole, this is also the historical reason we use ImageMagick, not PIL - the bash commands make it easier to make things user-customizable than if everything was based on purely Python code. But as a whole agree this adds some weight from spawning subprocesses.